### PR TITLE
[ODS-4401] Bring in Swashbuckle as a Replacement for swagger-ui

### PR DESCRIPTION
**Pull request by tbmeador-dlp**

### CODE REVIEW

File: Application/EdFi.Ods.SwaggerUI/ConfigureMetadata.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            _logger = logger;
+        }
+
+        public void Configure(SwaggerUIOptions options)
+        {
+            AddMetadataEndpoints(options);
+        }
+
+        private void AddMetadataEndpoints(SwaggerUIOptions options)
```
Comment (jamessmckay) [inline Application/EdFi.Ods.SwaggerUI/ConfigureMetadata.cs:31]: should we use async here

File: Application/EdFi.Ods.SwaggerUI/ConfigureMetadata.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            {
+                _logger.LogError($"'WebApiVersionUrl' property in appsettings.json was not found, skipping metadata retrieval.");
+                return;
+            }
+
+            try
+            {
+                using var httpClient = new HttpClient();
+
+                var versionResponse = httpClient
```
Comment (jamessmckay) [inline Application/EdFi.Ods.SwaggerUI/ConfigureMetadata.cs:43]: my concern is this, why not do it as an async task and then return the metadata json before the final enumeration

File: Application/EdFi.Ods.SwaggerUI/Program.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace EdFi.Ods.SwaggerUI
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
```
Comment (jamessmckay) [inline Application/EdFi.Ods.SwaggerUI/Program.cs:17]: should we add environment vars and the appsettings.json?

File: Application/EdFi.Ods.SwaggerUI/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                app.UseExceptionHandler("/Error");
+            }
+
+            app.Map("/appSettings.json", AppSettingsDelegate);
```
Comment (jamessmckay) [inline Application/EdFi.Ods.SwaggerUI/Startup.cs:71]: this is interesting,

File: Application/EdFi.Ods.SwaggerUI/appsettings.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+  "WebApiVersionUrl": "http://localhost:54746/",
```
Comment (jamessmckay) [inline Application/EdFi.Ods.SwaggerUI/appsettings.json:2]:
```suggestion
  "Urls" : "http://localhost:56641",
  "WebApiVersionUrl": "http://localhost:54746/",
```

Comment (jamessmckay) [reply Application/EdFi.Ods.SwaggerUI/appsettings.json:2]: note the urls field sets the main url that the application run on.

File: Application/EdFi.Ods.SwaggerUI/wwwroot/index.js [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+  var errorDescription = document.getElementById('errorDescription')
+  errorDescription.innerHTML += '<p>' + message + '</p>'
+  errorDescription.className = errorDescription.className.replace('hide', '') // IE11 support
+  errorDescription.classList.remove('hide')
+}
+
+// maps the json data from the metadata url to the sections variable
+function mapSections(json) {
+  json.forEach(function (data) {
+    var prefix = data.Prefix === '' ? 'Resources' : data.Prefix
```
Comment (jamessmckay) [inline Application/EdFi.Ods.SwaggerUI/wwwroot/index.js:65]: lets create a ticket to address this in the future when the netcore changes come in. Note the red in the image are the changes we will need to do.

image

File: Application/EdFi.Ods.SwaggerUI/Program.cs (hunk 3 above)
Comment (tbmeador-dlp) [reply Application/EdFi.Ods.SwaggerUI/Program.cs:17]: As discussed this is covered by the default builder.

**Pull request merged**